puppet-mode.el

```diff
@@ -632,6 +633,40 @@ of the initial include plus puppet-include-indent."
         (indent-line-to 0)))))
 
 
+;;;; SMIE
+
+(defvar puppet-use-smie nil
+  "Whether to use SMIE or not.")
```

Comment:
I don't think that we need this.  If SMIE works for us, we'll just require an Emacs version that's recent enough.  But it's too much effort to maintain to different indentation/navigation engines in this mode, imho.